src/cider/nrepl/middleware/reload.clj

```diff
@@ -0,0 +1,49 @@
+(ns cider.nrepl.middleware.reload
+  (:require
+   [clojure.main :refer [repl-caught]]
+   [clojure.string :as str]
+   [haystack.analyzer :as stacktrace.analyzer]
+   [nrepl.middleware.interruptible-eval :refer [*msg*]]
+   [nrepl.middleware.print :as print]
+   [nrepl.misc :refer [response-for]]
+   [nrepl.transport :as transport]
+   [orchard.misc :as misc]))
+
+(defn- user-reload
+  "clj-reload.core/reload from the user project.
+  Must be configured via clj-reload.core/init before being called."
+  []
+  (some-> (symbol "clj-reload.core" "reload")
+          resolve))
+
+(defn respond
+  [{:keys [transport] :as msg} response]
+  (transport/send transport (response-for msg response)))
+
+(defn- refresh-reply
+  [{:keys [::print/print-fn transport session id] :as msg}]
+  (let [{:keys [exec]} (meta session)]
+    (exec id
+          #(try
+             (let [reload (user-reload)]
+               (when-not reload
+                 (throw (ex-info "clj-reload.core/reload not found" {})))
```

Comment:
Looks like we don't need to be this stringent, following https://github.com/tonsky/clj-reload/commit/0a082e9de248e9b2440999da3ccd4c60780422e5

Reply:
Oh that's right! We can use the project one if available, and fall back to the bundled one if not.